lib/query/query_recurring_donations.rb

```diff
@@ -79,6 +84,13 @@ def self.full_search_expr(np_id, query)
       expr = expr.where("#{failed_or_active_clauses.join(' OR ')}")
     end
 
+    if include_last_failed_charge && query.key?(:started_at) && query.key?(:end_date)
+      expr = expr.where(
+        'failed_charges.created_at >= $started_at
+        AND failed_charges.created_at < $end_date',
+        { started_at: query[:started_at], end_date: query[:end_date] }
+      )
+    end
```

Comment:
Question: is there a situation in where we'd want to look for failed charges during a time period but not also use the same time period for the general filtering?

Reply:
I *THINK* they'd always be the same but I may be overlooking something.

Reply:
I don't think so, but I didn't follow what you think should be different.

Reply:
Oh, I get what's going on. There's not already a search filter for the recurring donations based upon date. I was thinking of the payments in queries.